File: Class_4/OOP/go/runner.go
package main

import (
	"fmt"
	"greeter/greeter"
)

func main() {
	var name string

	fmt.Print("Enter your name: ")
	fmt.Scanln(&name)

	provider, err := greeter.NewGreeterProvider(name)

	if err != nil {
		panic(err)
	}

	provider.Greet()
}

File: Class_4/OOP/php/runner.php
<?php

declare(strict_types=1);

require_once __DIR__ . '/GreeterProvider.php';

print("Enter your name: ");
$stdin = fopen('php://stdin', 'r');
$name = trim(fgets($stdin));
fclose($stdin);

$greeter = new GreeterProvider($name);
$greeter->greet();

$greeter->setGreeter(new TextGreeter());
$greeter->greet();
